activerecord/test/cases/scoping/default_scoping_test.rb

```diff
@@ -361,6 +361,10 @@ def test_unscoped_with_named_scope_should_not_have_default_scope
     assert_equal 10, DeveloperCalledJamis.unscoped { DeveloperCalledJamis.poor }.length
   end
 
+  def test_unscoped_with_joins_should_not_have_default_scope
+    assert_equal FirstPost.unscoped { Comment.joins(:first_post).to_a }, Comment.joins(:post).to_a
```

Comment:
This test is hard to follow. Why the unscoped version of `Comment.joins(:first_post).to_a` is equals to `Comment.joins(:post).to_a`. Should not the join be removed?